test/unit-tests/function/matrix/eigs.test.js

```diff
@@ -251,6 +266,8 @@ describe('eigs', function () {
       [4.24, -4.68, -3.33, 1.67, 2.80, 2.73],
       [4.14, 4.27, 3.05, 2.24, 2.73, -4.47]])
     const ans = eigs(H)
+    const justvalues = eigs(H, { eigenvectors: false })
+    assert.deepStrictEqual(ans.values, justvalues.values)
```

Comment:
This test still passes when I change `eigs(H, { eigenvectors: false })` to `eigs(H, { eigenvectors: true })`. 

Can you make sure the test actually verifies that `justvalues` does not contain eigenvectors?

Reply:
Hmm, why do we care whether the result of `eigs` with eigenvectors false definitely does _not_ contain eigenvectors? For some class of matrices, it might be just as fast to compute the eigenvectors as not (in fact, that's true for real symmetric matrices, I only turned the computation off because for very large matrices it uses noticeably more memory). As I see it, `eigenvectors: false` merely authorizes not computing the eigenvectors; it shouldn't _require_ they not be computed...

Reply:
I understand your point. 

I think it is important to have a predictable and consistent API. If the user specifies that he/she doesn't need `eigenvectors` in the output (hoping to improve on performance), I think it is best to simply not return it. Having the function _sometimes_ returning it despite configuring `{ eigenvectors: false }` is just hard to explain and asks for troubles. People may accidentally rely on the property when it is there for certain input (when `{ eigenvectors: false }` is defined elsewhere in the codebase, out of sight), and then code may break in production with different input when the `eigenvectors` is not returned. 
And you don't know what the user does with the output. Maybe the result is merged in some data structure using a spread operator, and then the unwanted `eigenvectors` property may propagate throughout the code base unwanted, causing issues.

If it doesn't make a difference for performance, it's totally fine if the function `eigs` internally calculates the `eigenvectors`. But if the user specifies that he/she doesn't want them in the output, let's just respect that and not return it.

Reply:
OK, all traces of eigenvectors expunged when `eigenvectors: false` option is specified.

Reply:
Ugh, that somehow made it fail code coverage. Will see how I can get more of complexEigs.js covered by the test...

Reply:
Well, hurray for codecov. The "balancing" code in complexEigs.js had some long-standing typos in it, so it was never firing (even though it should have been). So I have fixed the balancing code by making it agree with http://www.phys.uri.edu/nigh/NumRec/bookfpdf/f11-5.pdf, and now it indeed fires on a couple of our test matrices. Ironically enough, that made the precision slightly worse on one really bad matrix we're testing. But that doesn't fuss me too much: our current methods are basically inadequate on that matrix, so there's not much we can do with it until we either switch to a more mature implementation or decide to bite the bullet and implement more current methods ourselves. Anyhow, the coverage is much better now, and all tests are passing with just a slightly relaxed test in that one case (note that before #3037, that case didn't work _at all_, so we weren't even checking those eigenvalues, so the tiny reduction in precision by putting in balancing is no indication of a problem. Both before and after enabling balancing, the eigenvalues are still quite crude compared to the true eigenvalues for that matrix.) So I think we are OK to move ahead with this PR.

Reply:
Thanks Glen. Great that you found and tackled this other issue in `eigs` regarding the balancing code!